homeassistant/components/daikin/number.py

```diff
@@ -0,0 +1,146 @@
+        self._attr_native_min_value = self._target_temperature - 2
+        self._attr_native_max_value = self._target_temperature + 2
+
+    @property
+    def native_value(self) -> float:
+        """Return the state of the sensor."""
+        if self.has_updated_value():
+            self._previous_value = None
+            self._current_value = self._api.device.zones[self._zone_id][2]
+        return self._current_value
+
+    def has_updated_value(self) -> bool:
+        """Detect if api has an updated value."""
+        if self._previous_value == self._api.device.zones[self._zone_id][2]:
+            return False
+        if self.value_in_range(self._api.device.zones[self._zone_id][2]):
+            return True
+        return False
+
+    def value_in_range(self, value: float) -> bool:
+        """Check temperature is within range."""
+        if value < (self._target_temperature - 2) or value > (
+            self._target_temperature + 2
+        ):
+            return False
+        return True
+
+    async def async_added_to_hass(self) -> None:
+        """Add listener when entity about to be added to hass."""
+        registry = self.async_get_registry()
+        entity_id = registry.async_get_entity_id(
+            Platform.CLIMATE, DAIKIN_DOMAIN, self._api.device.mac
+        )
+
+        if entity_id is not None:
+            self._remove_listener = async_track_state_change_event(
+                self.hass, entity_id, self.async_update_value
+            )
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Remove listener when entity about to be removed to hass."""
+        if self._remove_listener is not None:
+            self._remove_listener()
+
+    @callback
+    def async_update_value(self, event: Event) -> None:
+        """Update zone state."""
+        new_state = event.data.get("new_state")
+        if new_state is None:
+            return None
+
+        if (new_temp := new_state.attributes.get(ATTR_TEMPERATURE)) is None:
+            return None
+
+        self._target_temperature = new_temp
+
+        if not self.value_in_range(self._current_value):
+            self._previous_value = self._current_value
+
+            if self._current_value < (new_temp - 2):
+                self._current_value = new_temp - 2
+            elif self._current_value > (new_temp + 2):
+                self._current_value = new_temp + 2
+
+        self.async_write_ha_state()
+
+    @callback
+    def async_get_registry(self) -> er.EntityRegistry:
+        """Get or return Entity Registry."""
+        if self._registry is None:
+            self._registry = er.async_get(self.hass)
+        return self._registry
+
+    async def async_update(self) -> None:
+        """Retrieve latest state."""
+        await self._api.async_update()
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set the zone temperature."""
+        if not self.value_in_range(value):
+            return None
```

Comment:
Should raise `HomeAssistantError` so we can display a message to the user and not just silently abort